### PR TITLE
CRB 259/minimum difficulty

### PR DESCRIPTION
**Pull request by CAGS295**
depends on #12

### CODE REVIEW

File: src/engine.rs
Comment (nathanwhit) [inline src/engine.rs]: I'm confused by this comment, the types of the error _do_ match up. Replacing this statement with just 
```rust
node.initialize(startup)?;
Ok(node)
```
compiles and runs correctly.

File: src/miner/miner.rs
Comment (nathanwhit) [inline src/miner/miner.rs]: I'm a little confused by "block head header" is it a typo, or what is a "block head"?

**Review by nathanwhit — COMMENTED** [review]
A couple comments but overall looks good to me. Should be good to go once #10 is merged.

File: src/engine.rs
Comment (CAGS295) [reply src/engine.rs]: Good catch, these are old comments.

File: src/miner/miner.rs
Comment (CAGS295) [reply src/miner/miner.rs]: Yup, that's not very descriptive. What about Chain head header?

**Review by nathanwhit — DISMISSED** [review]
Looks good to me

**Review by atodorov — CHANGES_REQUESTED** [review]
Needs rebase b/c there are some conflicts.

**Review by atodorov — APPROVED** [review]
After discussion at https://gluwa.slack.com/archives/C7JJP9WTF/p1637333833199600 we agreed to merge this PR.

**Review by nathanwhit — APPROVED** [review]

**Pull request merged**
